### PR TITLE
doc(unique task names howto): add Nico's feedback

### PR DESCRIPTION
**Pull request by NathalieC**
change scripts names
homogenize task name Validate request
fix typos
add recommendation to add description to contract inputs
add complete script content when editing business data default value

### CODE REVIEW

File: md/optimize-user-tasklist.md [hunk 1]
```diff
@@ -50,11 +50,13 @@ To make sure the process gets the information it needs to start a new instance,
 3. Click **Next**
 4. Unselect **requesterName** and **status** as the process doesn't need this information from the form to instantiate  
    The default values of business data attributes mapped to contract inputs are automatically set thanks to a script generated along with the contract inputs themselves.  
-5. Click **Preview** to view this script, and then **Finish**, **OK** and **OK**
+5. Click **Finish**, carefully read the warnings, and press **OK** and **OK**.
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: yeah, right...

File: md/optimize-user-tasklist.md [hunk 2]
```diff
@@ -50,11 +50,13 @@ To make sure the process gets the information it needs to start a new instance,
...
 6. In the **Execution** pane > **Contract** tab, a complex contract input is created, mapped to the selected attributes of the **leaveRequest** business variable, as shown here:
 
 ![Contract](images/contract_for_tasklist.png)<!--{.img-responsive .img-thumbnail}-->
 
+In actual BPM projects, we recommend you also add a description to each contract input. It will be used as input field caption for end-users in the auto-generated form, if you decide to use such forms up to the User Acceptance Test phase of your project.
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: missing a `to` right?

File: md/optimize-user-tasklist.md [hunk 3]
```diff
@@ -151,14 +164,15 @@ When reusing these features for your own projects, make sure that the result of
...
 1. Save the diagram
-2. Run the pool
+2. Run the pool. The default loggued user is Walter Bates
 3. Fill out the default instantiation form with two dates
 4. Click on **Start**  
 
 Once you are sent to the task list, you can see that the task name is contextualized.  
 It is computed once, when the task becomes ready. 
 
 **Note:** The following set of instructions only applies to the task list in Bonita BPM 7.3.0 and above. In earlier versions, the **Description** field is displayed by default.
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: does it need some 
```
::: info
:::
```
block ?

File: md/optimize-user-tasklist.md [hunk 4]
```diff
@@ -65,7 +67,18 @@ When the leave request is submitted, _status_ and _requesterName_ must default v
     def initiator = BonitaUsers.getProcessInstanceInitiator(apiAccessor,processInstanceId);
     leaveRequestVar.requesterName = "$initiator.firstName $initiator.lastName"
 ```
-5. Click **OK** and **OK**
+so it becomes:
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: indent it to be aligned with the ordered list ?

File: md/optimize-user-tasklist.md [hunk 5]
```diff
@@ -98,19 +111,19 @@ As a good practice, we advise to display static information that defines the tas
...
 The task display name will be made of the requester's first name, last name, and the leave start date and end date.
 1. Next to the **Display name** field, click the **pencil** icon to display the expression editor.
 2. Select the **Script** expression
-3. Give it a name: **Validation display name**
+3. Give it a name: **buildValidateRequestDisplayName()**
 4. Type the script. 
 
 ```groovy
-return "Leave validation for ${leaveRequest.requesterName}: ${leaveRequest.startDate.format('yyyy-M-d')} / ${leaveRequest.endDate.format('yyyy-M-d')}".toString()
+return "Validate leave: ${leaveRequest.requesterName}: ${leaveRequest.startDate.format('yyyy-M-d')} / ${leaveRequest.endDate.format('yyyy-M-d')}".toString()
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: indent it to be aligned with the ordered list ?

File: md/optimize-user-tasklist.md [hunk 6]
```diff
@@ -120,11 +133,11 @@ return "Leave validation for ${leaveRequest.requesterName}: ${leaveRequest.start
 As a dynamic information, status is addressed here:
 1. Next to the **Display description** field, click the **pencil** icon to display the expression editor.
 2. Select the **Script** expression type
-3. Give it a name: **Validation display description**
+3. Give it a name: **buildValidateRequestDisplayDescription()**
 4. Type the script. 
 
 ```groovy
-return "A leave request has been ${leaveRequest.status}".toString()
+return "${leaveRequest.status}".toString()
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: indent it to be aligned with the ordered list ?

File: md/optimize-user-tasklist.md [hunk 7]
```diff
@@ -134,14 +147,14 @@ return "A leave request has been ${leaveRequest.status}".toString()
 The status of the request when it's validated as well as the validator's name are displayed in the **Done tasks** filter of the user task list. To define it:
 1. Next to the **Description after completion** field, click the **pencil** icon to display the expression editor.
 2. Select the **Script** expression type
-3. Give it a name: **Validation desc after completion**
+3. Give it a name: **buildValidateRequestDescAfterCompletion()**
 4. Type the script.
 
 ```groovy
```
Comment (redboul) [inline md/optimize-user-tasklist.md]: indent it to be aligned with the ordered list ?

File: md/optimize-user-tasklist.md (hunk 2 above)
Comment (NathalieC) [inline md/optimize-user-tasklist.md]: i can add one "we recommend you to also add". The sentence is correct without, but yeah, it's easier to understand with it.

File: md/optimize-user-tasklist.md (hunk 3 above)
Comment (NathalieC) [inline md/optimize-user-tasklist.md]: I thought it was too big an emphasis. Those wells are real attention focus traps. We should use them with care.

**Comment by tnguyen1** [comment]
retest this please

**Pull request merged**